omniqueue/tests/it/azure_queue_storage.rs

```diff
@@ -0,0 +1,262 @@
+use std::{
+    collections::HashSet,
+    time::{Duration, Instant},
+};
+
+use azure_storage::StorageCredentials;
+use azure_storage_queues::QueueServiceClientBuilder;
+use omniqueue::{
+    backends::{AqsBackend, AqsConfig, AqsConsumer, AqsProducer},
+    QueueBackend, QueueError,
+};
+use serde::{Deserialize, Serialize};
+
+async fn create_queue_get_a_pair() -> (AqsProducer, AqsConsumer) {
+    create_queue_get_a_pair_with_receive_timeout(None).await
+}
+
+async fn create_queue_get_a_pair_with_receive_timeout(
+    receive_timeout: Option<Duration>,
+) -> (AqsProducer, AqsConsumer) {
+    let queue_name: String = std::iter::repeat_with(fastrand::lowercase)
+        .take(8)
+        .collect();
+
+    let credentials = StorageCredentials::access_key(
+        azure_storage::EMULATOR_ACCOUNT.to_string(),
+        azure_storage::EMULATOR_ACCOUNT_KEY.to_string(),
+    );
+    let cfg = AqsConfig {
+        queue_name,
+        empty_receive_delay: None,
+        message_ttl: Duration::from_secs(90),
+        storage_account: azure_storage::EMULATOR_ACCOUNT.to_string(),
+        credentials: credentials.clone(),
+        cloud_uri: Some(format!(
+            "http://localhost:10001/{}",
+            azure_storage::EMULATOR_ACCOUNT
+        )),
+        receive_timeout,
+    };
+
+    let cli = QueueServiceClientBuilder::new(cfg.storage_account.clone(), credentials)
+        .cloud_location(azure_storage::CloudLocation::Custom {
+            account: cfg.storage_account.clone(),
+            uri: cfg.cloud_uri.clone().unwrap(),
+        })
+        .build()
+        .queue_client(cfg.queue_name.clone());
+
+    cli.create().into_future().await.unwrap();
+
+    AqsBackend::new_pair(cfg).await.unwrap()
+}
+
+#[derive(Debug, Deserialize, Serialize, Eq, Hash, PartialEq)]
+pub struct ExType {
+    a: String,
+}
+
+#[tokio::test]
+async fn test_raw_send_recv() {
+    let (producer, mut consumer) = create_queue_get_a_pair().await;
+
+    let payload = "test123";
+    producer.send_raw(payload).await.unwrap();
+
+    let mut d = consumer.receive().await.unwrap();
+    assert_eq!(
+        payload,
+        &String::from_utf8(d.take_payload().unwrap()).unwrap()
+    );
+    d.ack().await.unwrap();
+}
+
+#[tokio::test]
+async fn test_serde_send_recv() {
+    let (producer, mut consumer) = create_queue_get_a_pair().await;
+
+    let payload = ExType {
+        a: "test123".to_string(),
+    };
+    producer.send_serde_json(&payload).await.unwrap();
+
+    let d = consumer.receive().await.unwrap();
+    assert_eq!(d.payload_serde_json::<ExType>().unwrap().unwrap(), payload);
+    d.ack().await.unwrap();
+}
+
+// Note: Azure Queue Storage doesn't guarantee order of messages, hence
+// the HashSet popping and length validation instead of assuming
+// particular values:
```

Comment:
Same for this, would be nice to have it in the documentation (at least of the backend type and/or the module).

Reply:
I'm not sure this particular note needs to be documented in our crate. This is just how AQS works. Same considerations apply to SQS as well (possibly others)

Reply:
But does it hurt to document it? Anyways, not hung up on this so if you prefer to leave it out, leave it out.

Reply:
Ok, added.